src/components/DeleteAccountModal/DeleteAccountModal.tsx

```diff
@@ -0,0 +1,101 @@
+import { useTranslation } from 'react-i18next';
+import { type SchemaOf, object, string } from 'yup';
+import { useNavigate, useLocation } from 'react-router';
+import { useState } from 'react';
+import { useMutation } from 'react-query';
+
+import PasswordField from '../PasswordField/PasswordField';
+import Button from '../Button/Button';
+
+import styles from './DeleteAccountModal.module.scss';
+
+import type { DeleteAccountFormData } from '#types/account';
+import useForm from '#src/hooks/useForm';
+import { addQueryParam, removeQueryParam } from '#src/utils/location';
+import { deleteAccountData, logout } from '#src/stores/AccountController';
+
+const DeleteAccountModal = () => {
+  const { t } = useTranslation('user');
+
+  const [enteredPassword, setEnteredPassword] = useState<string>('');
+
+  const deleteAccount = useMutation(deleteAccountData, {
+    onSuccess: async () => {
+      await logout();
+      navigate('/');
+    },
+    onError: () => {
+      setEnteredPassword('');
+      handleCancel();
+    },
+  });
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const validationSchema: SchemaOf<DeleteAccountFormData> = object().shape({
+    password: string().required(t('login.field_required')),
+  });
+  const initialValues: DeleteAccountFormData = { password: '' };
+  const { handleSubmit, handleChange, values, errors } = useForm(
+    initialValues,
+    () => {
+      setEnteredPassword(values.password);
+      navigate(addQueryParam(location, 'confirmation', 'true'));
+    },
+    validationSchema,
+  );
+
+  const handleCancel = () => {
+    const removedU = removeQueryParam(location, 'u');
+    navigate(removedU.split('&confirmation')[0]);
+  };
+
+  return enteredPassword ? (
+    <div className={styles.formContainer}>
+      <h2 className={styles.heading}>{t('account.delete_account_title')}</h2>
+      <div className={styles.innerContainer}>
+        <p className={styles.paragraph}>{t('account.delete_account_modal_text_1')}</p>
+        <p className={styles.paragraph}>{t('account.delete_account_modal_text_2')}</p>
```

Comment:
Can we use `\n` in the translation?

Reply:
I had issues with trying to use `\n` and break tags in the translation so decided to split them up in separate html elements